lm_eval/tasks/wino_bias.py

```diff
@@ -0,0 +1,108 @@
+
+Winograd-schema evaluation of gendered coreference resolution.
+The dataset contains pro-stereotypical and anti-stereotypical parts. The difference in accuracy for those two subsets
+quatnifies bias.
+
+Homepage: https://uclanlp.github.io/corefBias/overview
+"""
+from lm_eval.base import PromptSourceTask, mean
+import transformers.data.metrics.squad_metrics as squad_metrics
+
+_CITATION = """
+@inproceedings{zhao-etal-2018-gender,
+    title = "Gender Bias in Coreference Resolution: Evaluation and Debiasing Methods",
+    author = "Zhao, Jieyu  and
+      Wang, Tianlu  and
+      Yatskar, Mark  and
+      Ordonez, Vicente  and
+      Chang, Kai-Wei",
+    booktitle = "Proceedings of the 2018 Conference of the North {A}merican Chapter of the Association for Computational Linguistics: Human Language Technologies, Volume 2 (Short Papers)",
+    month = jun,
+    year = "2018",
+    address = "New Orleans, Louisiana",
+    publisher = "Association for Computational Linguistics",
+    url = "https://aclanthology.org/N18-2003",
+    doi = "10.18653/v1/N18-2003",
+    pages = "15--20",
+    abstract = "In this paper, we introduce a new benchmark for co-reference resolution focused on gender bias, WinoBias. Our corpus contains Winograd-schema style sentences with entities corresponding to people referred by their occupation (e.g. the nurse, the doctor, the carpenter). We demonstrate that a rule-based, a feature-rich, and a neural coreference system all link gendered pronouns to pro-stereotypical entities with higher accuracy than anti-stereotypical entities, by an average difference of 21.1 in F1 score. Finally, we demonstrate a data-augmentation approach that, in combination with existing word-embedding debiasing techniques, removes the bias demonstrated by these systems in WinoBias without significantly affecting their performance on existing datasets.",
+}
+"""
+
+
+class WinoBias(PromptSourceTask):
+    VERSION = 0
+    DATASET_PATH = "wino_bias"
+
+    def has_training_docs(self):
+        return False
+
+    def has_validation_docs(self):
+        return True
+
+    def has_test_docs(self):
+        return True
+
+    def training_docs(self):
+        pass
+
+    def validation_docs(self):
+        return self.dataset["validation"]
+
+    def test_docs(self):
+        return self.dataset["test"]
+
+    def stopping_criteria(self):
+        return "\n"
+
+    def process_results(self, doc, results):
+        """Take a single document and the LM results and evaluates, returning a
+        dict where keys are the names of submetrics and values are the values of
+        the metric for that one document
+
+        :param doc:
+            The document as returned from training_docs, validation_docs, or test_docs.
+        :param results:
+            The results of the requests created in construct_requests.
+        """
+        target = self.doc_to_target(doc).strip()
+        pred = " ".join(results[0].strip().split(" ")[:len(target.split(" "))])
```

Comment:
Hi, @tomlimi ! Is it standard here to truncate predictions to the target (n-gram) length? Seems reasonable for this task but wanted to double check.

Reply:
afair I didn't want to penalize the output for being longer, for instance, when it's more specific. Setting `max_generation_length` to ~5 could also slove it . I remember that when I was working on that `max_generation_length` was buggy, that's why I did it this way.

Reply:
@tomlimi we report both actual accuracy and length-normalized accuracy to help compensate for this

Reply:
@tomlimi The byte-length normalized accuracy metric was not integrated with this branch. It's now been added [here](https://github.com/bigscience-workshop/lm-evaluation-harness/blob/257bfb420bc0811a0f7fd791240f6dc179350899/lm_eval/base.py#L737)  and an example can be found within the `process_results` method of `superglue.py`'s [` WinogenderSchemaDiagnostics` ](https://github.com/bigscience-workshop/lm-evaluation-harness/blob/257bfb420bc0811a0f7fd791240f6dc179350899/lm_eval/tasks/superglue.py#L316).

Reply:
Thank you @jon-tow ! In WinoGender, we have `answer_choices_list` so I see that the scores are normalized by the size of the choices, great idea. Here the task is solved by text generating, and then the exact match is reported. So normalization wouldn't be applicable here, am I right?

Reply:
Oh, yup! You're correct this would not be applicable; sorry!